### ISSUE TITLE
Wrong download URL for GraalVM 19.0.0

### ISSUE THREAD

**Issue by vard-io** [palantir/gradle-graal#105]
```
OS: macOS 10.14.5 (18F132) 
Gradle Version: 5.4.1
Java Version: 8.0.212-zulu
```

I just tried to use this Gradle plugin to build a GraalVM native image with the 19.0.0 GraalVM release. Unfortunately it seems the Download URLs have changed a little and thus the Plugin can not download the GraalVM release from GitHub.

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':downloadGraalTooling'.
> java.io.FileNotFoundException: https://github.com/oracle/graal/releases/download/vm-19.0.0/graalvm-ce-19.0.0-macos-amd64.tar.gz
```

The correct download URL in my case would be `https://github.com/oracle/graal/releases/download/vm-19.0.0/graalvm-ce-darwin-amd64-19.0.0.tar.gz`

A simple and future proof solution would be to let the user specify the exact `downloadUrl`, which overwrites the `downloadBaseUrl` and `graalVersion` settings.

Additionally the way the URLs are built from `downloadBaseUrl` and `graalVersion` should be adjusted, as I expect this problem to persist for the upcoming releases of GraalVM.

**Comment by Yegair** [comment]
Seems to have already been fixed, but not merged yet: https://github.com/palantir/gradle-graal/pull/104

**Comment by albertoclarit** [comment]
how did you guys solved it

**Comment by juddgaddie** [comment]
@albertoclarit  When the PR #104 is merged and released then it will be solved.

**Comment by mikaelhg** [comment]
You can work around the URL pattern issue by setting the `downloadBaseUrl` in your `build.gradle` to the actual TGZ URL, and ending it with a `?` or a `#`. That will only help you with the URL, though, and not with the changed file structure in the release version.

**Comment by yschimke** [comment]
@mikaelhg I don't think the changed internal structure is breaking anything with the plugin.

```
diff --git a/okcurl/build.gradle b/okcurl/build.gradle
index d10565b65..c4743a344 100644
--- a/okcurl/build.gradle
+++ b/okcurl/build.gradle
@@ -39,7 +39,8 @@ dependencies {
 }

 graal {
-  graalVersion("1.0.0-rc16")
+  graalVersion("19.0.2")
+  downloadBaseUrl("https://github.com/oracle/graal/releases/download/vm-19.0.2/graalvm-ce-darwin-amd64-19.0.2.tar.gz?a=")
   mainClass("okhttp3.curl.Main")
   outputName("okcurl")
   option("--configurations-path")
```

**Comment by juddgaddie** [comment]
@yschimke your diff above pulls down the correct graal but native-image is no longer included in the Graal release so it fails. Is this what you expect ?

**Comment by yschimke** [comment]
@juddgaddie yeah, I fooled myself.  It runs and fails because its missing native-image, but I repaired it.  

Running `~/.gradle/caches/com.palantir.graal/19.0.2/graalvm-ce-19.0.2/Contents/Home/bin/gu install native-image` makes it work forevermore.

Sorry, should have updated here.

**Comment by juddgaddie** [comment]
@yschimke thanks, I should really fix the plugin. I would want to know from the maintainers that they would take a fix and release it?

**Comment by uschi2000** [comment]
Of course, any time.

**Comment by mageddo** [comment]
> You can work around the URL pattern issue by setting the `downloadBaseUrl` in your `build.gradle` to the actual TGZ URL, and ending it with a `?` or a `#`. That will only help you with the URL, though, and not with the changed file structure in the release version.

What a creative hack, thanks

**Comment by mikaelhg** [comment]
It's really unfortunate that so many plugin class methods are pointlessly marked `final`, because otherwise extending the plugin to work around these kinds of issues would be a piece of cake.

**Comment by juddgaddie** [comment]
@mikaelhg how much harder would it be to fix the plugin?

**Comment by StevenACoffman** [comment]
Uh... unless I am missing something, to "fix" it on extract [right here](https://github.com/palantir/gradle-graal/blob/develop/src/main/java/com/palantir/gradle/graal/ExtractGraalTask.java#L63) the plugin would always want to then exec `./gu install native-image`, right?

**Comment by markelliot** [comment]
Sorry for the delay, but this should now be resolved in 0.4.0.
